### PR TITLE
[Feature-companion] Add renter to metadata:extra

### PR DESCRIPTION
**Pull request by hmoco**
Needs to be merged with https://github.com/CenterForOpenScience/osf.io/pull/3724

### CODE REVIEW

**Comment by samanehsan** [comment]
I took a look at this along with https://github.com/CenterForOpenScience/osf.io/pull/3724 and it looks good to me! @hmoco :whale:

File: tests/providers/osfstorage/test_provider.py [hunk 1]
```diff
@@ -112,6 +112,7 @@ def upload_response():
             'downloads': 10,
             'version': 8,
             'path': '/dfl893b1pdn11kd28b',
+            'checkout': '',
```
Comment (chrisseto) [inline tests/providers/osfstorage/test_provider.py]: Use a non-falsey value here

**Comment by chrisseto** [comment]
LGTM, ready to merge whenever @sloria merges the OSF side.

**Comment by chrisseto** [comment]
Merged via https://github.com/CenterForOpenScience/waterbutler/commit/f5f8cb091ea967595667686b6c499736b6a214a5

**Pull request closed without merging**